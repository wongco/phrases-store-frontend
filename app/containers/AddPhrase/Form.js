import styled from 'styled-components';

const Form = styled.form`
  display: flex;
  flex-direction: row;
  justify-content: center;
  width: 80%;
  margin-bottom: 1em;
`;

export default Form;
