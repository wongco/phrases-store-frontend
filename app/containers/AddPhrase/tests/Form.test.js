import React from 'react';
import { shallow, mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import toJson from 'enzyme-to-json';
import Form from '../Form';

describe('<Form />', () => {
  it('renders without crashing', () => {
    shallow(<Form />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<Form />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const id = 'testId';
    const wrapper = mount(<Form id={id} />);
    const renderedComponent = enzymeFind(wrapper, Form);
    expect(renderedComponent.prop('id')).toBe('testId');
  });

  it('should render its text', () => {
    const children = 'Text';
    const wrapper = mount(<Form>{children}</Form>);
    const renderedComponent = enzymeFind(wrapper, Form);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});
