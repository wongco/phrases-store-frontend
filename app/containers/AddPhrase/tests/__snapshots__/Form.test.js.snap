// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Form-dg7sp0-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;flex-direction:row;justify-content:center;width:80%;margin-bottom:1em;",
        ],
      },
      "displayName": "Form",
      "render": [Function],
      "styledComponentId": "Form-dg7sp0-0",
      "target": "form",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
